cle/backends/pe.py

```diff
@@ -55,35 +57,33 @@ def relocate(self, solist, bypass_compatibility=False):
                 # no work required
                 pass
             elif self.reloc_type == pefile.RELOCATION_TYPE['IMAGE_REL_BASED_HIGHLOW']:
-                org_bytes = ''.join(self.owner_obj.memory.read_bytes(self.addr, 4))
+                org_bytes = ''.join(self.owner_obj.memory.read_bytes(
+                    AT.from_lva(self.addr, self.owner_obj).to_rva(), 4))
                 org_value = struct.unpack('<I', org_bytes)[0]
-                rebased_value = org_value + self.owner_obj.rebase_addr - self.owner_obj.requested_base
-                rebased_bytes = struct.pack('<I', rebased_value % 2**32)
-                self.owner_obj.memory.write_bytes(self.dest_addr, rebased_bytes)
+                rebased_value = AT.from_lva(org_value, self.owner_obj).to_mva()
+                rebased_bytes = struct.pack('<I', rebased_value % 2 ** 32)
+                self.owner_obj.memory.write_bytes(
+                    AT.from_lva(self.dest_addr, self.owner_obj).to_rva(), rebased_bytes)
             elif self.reloc_type == pefile.RELOCATION_TYPE['IMAGE_REL_BASED_DIR64']:
-                org_bytes = ''.join(self.owner_obj.memory.read_bytes(self.addr, 8))
+                org_bytes = ''.join(self.owner_obj.memory.read_bytes(
+                    AT.from_lva(self.addr, self.owner_obj).to_rva(), 8))
                 org_value = struct.unpack('<Q', org_bytes)[0]
-                rebased_value = org_value + self.owner_obj.rebase_addr - self.owner_obj.requested_base
+                rebased_value = AT.from_lva(org_value, self.owner_obj).to_mva()
                 rebased_bytes = struct.pack('<Q', rebased_value)
-                self.owner_obj.memory.write_bytes(self.dest_addr, rebased_bytes)
+                self.owner_obj.memory.write_bytes(AT.from_lva(self.dest_addr, self.owner_obj).to_rva(), rebased_bytes)
             else:
                 l.warning('PE contains unimplemented relocation type %d', self.reloc_type)
         else:
             return super(WinReloc, self).relocate(solist, bypass_compatibility)
 
+
 class PESection(Section):
     """
     Represents a section for the PE format.
     """
-    def __init__(self, pe_section):
-        super(PESection, self).__init__(
-            pe_section.Name,
-            pe_section.Misc_PhysicalAddress,
-            pe_section.VirtualAddress,
-            pe_section.Misc_VirtualSize,
-        )
-
-        self.characteristics = pe_section.Characteristics
+    def __init__(self, name, offset, vaddr, size, chars):
+        super(PESection, self).__init__(name, offset, vaddr, size)
+        self.characteristics = chars
```

Comment:
If you feel strongly about this change, could you please update ELFSection and ELFSegment to do the same thing? Right now they both just take readelf sections and parse them out into the constructors for the normal region arguments; this class was designed to mirror that. If there are complications with that for some reason, please revert this change. The interfaces should be the same.